students/eric_grandeo/lesson02/charges_calc.py

```diff
@@ -0,0 +1,126 @@
+
+def calculate_additional_fields(data):
+    """Checking values in JSON, calculating total days,
+       total price, sqrt total price, & unit cost"""
+
+    for value in data.values():
+        try:
+            logging.info("Called with value: {}".format(value))
+
+            #logs a warning if no rental end date is available
+            if value['rental_start'] == '':
+                logging.warning("Missing rental start date")
+            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            #logs the rental start date
+            logging.debug('Rental start: {}'.format(value['rental_start']))
+
+            #logs a warning if no rental end date is available
+            if value['rental_end'] == '':
+                logging.warning("Missing rental end date")
+            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+            #logs the rental end date
+            logging.debug('Rental ends: {}'.format(value['rental_end']))
+
+            value['total_days'] = (rental_end - rental_start).days
+            #logs total days
+            logging.debug('Total days: {:d}'.format(value['total_days']))
+
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            #logs the total price
+            logging.debug('Total price: ${:,.2f}'.format(value['total_price']))
+
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            #logs square root of total price
+            logging.debug('Square root of total price: {:06.2f}'.format(value['sqrt_total_price']))
+
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+            logging.debug('Unit cost: ${:,.2f}'.format(value['unit_cost']))
+        except ValueError:
+            #logs an error if total days are negative, resulting in attempting to find square
+            #root of a negative number
+            logging.error("Total days must not be negative; end date is before start date")
+
+    return data
+
+def save_to_json(filename, data):
+    """Save new Values to JSON file"""
+
+    with open(filename, 'w') as file:
+        json.dump(data, file)
+
+if __name__ == "__main__":
+    ARGS = parse_cmd_arguments()
+    if ARGS.debug == '0':
+        LOGGER.disabled = True
+    elif ARGS.debug == '1':
+        FILE_HANDLER.setLevel(logging.ERROR)
+        CONSOLE_HANDLER.setLevel(logging.ERROR)
+        LOGGER.setLevel(logging.ERROR)
+    elif ARGS.debug == '2':
+        FILE_HANDLER.setLevel(logging.WARNING)
+        CONSOLE_HANDLER.setLevel(logging.WARNING)
+        LOGGER.setLevel(logging.WARNING)
+    elif ARGS.debug == '3':
+        #From assignment: Debug: General comments, indicating where in the script flow we are.
+        #Should be shown on screen only (i.e., never saved to logfile).
+        FILE_HANDLER.setLevel(logging.WARNING)
+        CONSOLE_HANDLER.setLevel(logging.DEBUG)
+        LOGGER.setLevel(logging.DEBUG)
+
```

Comment:
Hi Eric,

Right here, you need to add some code to error out if an invalid level is entered (for example: -d 10). Please submit a new pull request once you've made the change.

Regards,

Luis